crates/bevy_ecs/src/system/mod.rs

```diff
@@ -517,7 +519,49 @@ mod tests {
     }
 
     #[test]
-    fn removal_tracking() {
+    fn iterate_components() {
+        let mut world = World::default();
+        world.insert_resource(Vec::<String>::new());
+
+        world.spawn().insert(A).insert(B);
+
+        world.spawn().insert(A).insert(C);
+
+        fn sys(components: Components, mut names: ResMut<Vec<String>>) {
+            for component_info in &components {
+                names.push(component_info.name().to_string());
+            }
+        }
+
+        run_system(&mut world, sys);
+        assert_eq!(world.get_resource::<Vec<String>>().unwrap().len(), 3);
+    }
+
+    #[test]
+    fn iterate_resources() {
+        let mut world = World::default();
+        world.insert_resource(Vec::<String>::new());
+        world.insert_resource(42usize);
+
+        world.spawn().insert(A).insert(B);
+
+        world.spawn().insert(A).insert(C);
+
+        fn sys(resources: Resources, mut names: ResMut<Vec<String>>) {
+            for resource_info in &resources {
+                names.push(resource_info.name().to_string());
+            }
+        }
+
+        run_system(&mut world, sys);
+        assert!(world
+            .get_resource::<Vec<String>>()
+            .unwrap()
+            .contains(&"usize".to_string()),);
+    }
+
+    #[test]
+    fn remove_tracking() {
```

Comment:
```suggestion
    fn removal_tracking() {
```

Just to reduce diff noise.